youtube_mcp.py

```diff
@@ -0,0 +1,449 @@
+"""
+YouTube Download MCP Server using token.pickle authentication.
+
+This MCP server provides tools for accessing YouTube videos owned by 
+the resource owner using token.pickle for authentication.
+"""
+
+import base64
+import os
+import pickle
+from pathlib import Path
+from typing import Any, Optional
+
+from google.auth.transport.requests import Request
+from google.oauth2.credentials import Credentials
+from googleapiclient.discovery import build
+from mcp.server import FastMCP
+
+# Try to import Playwright (optional)
+try:
+    from playwright.sync_api import sync_playwright, TimeoutError as PlaywrightTimeoutError
+    import pyotp
+    PLAYWRIGHT_AVAILABLE = True
+except ImportError:
+    PLAYWRIGHT_AVAILABLE = False
+
+# Initialize FastMCP server
+mcp = FastMCP("YouTube Downloader")
+
+def get_authenticated_service():
+    """
+    Load credentials from token.pickle (file or environment variable) and return authenticated YouTube service.
+    
+    Supports two methods:
+    1. TOKEN_PICKLE_BASE64 environment variable containing base64-encoded pickle data
+    2. token.pickle file in current directory
+    
+    Returns:
+        Authenticated YouTube API service
+    """
+    creds = None
+    
+    # Try loading from environment variable first
+    token_pickle_b64 = os.environ.get('TOKEN_PICKLE_BASE64')
+    if token_pickle_b64:
+        try:
+            pickle_data = base64.b64decode(token_pickle_b64)
+            creds = pickle.loads(pickle_data)
+        except Exception as e:
+            raise ValueError(f"Failed to load credentials from TOKEN_PICKLE_BASE64: {e}")
+    elif os.path.exists("token.pickle"):
+        # Fall back to file-based loading
+        with open("token.pickle", 'rb') as token:
+            creds = pickle.load(token)
+    else:
+        raise ValueError("No token.pickle found. Set TOKEN_PICKLE_BASE64 env var or provide token.pickle file.")
+    
+    # Refresh credentials if expired
+    if creds and creds.expired and creds.refresh_token:
+        creds.refresh(Request())
+        # Save refreshed credentials back if using file
+        if not token_pickle_b64 and os.path.exists("token.pickle"):
+            with open("token.pickle", 'wb') as token:
+                pickle.dump(creds, token)
+    
+    if not creds or not creds.valid:
+        raise ValueError("Invalid credentials. Please authenticate first.")
+    
+    return build('youtube', 'v3', credentials=creds)
+
+
+@mcp.tool()
+def download_video(
```

Comment:
Name is misleading. Maybe get download URL or similar. Reason about the naming choices of these functions and make changes accordingly